From: RSDK-iOS-Builds <builds@rsdk.dev>
Subject: [PATCH] RSDKv1 iOS platform support

Add RETRO_PLATFORM == RETRO_iOS support to RSDKv2 decompilation
(adapted for RSDKv1 data format).

Based on danielgpinheiro/RSDKv1-xbox (Original Xbox port via nxdk).
Adaptado do port de Original Xbox do danielgpinheiro (nxdk).
https://github.com/danielgpinheiro/RSDKv1-xbox

---
 RSDKv2/RetroEngine.hpp  |  6 ++++++
 RSDKv2/Userdata.cpp     |  7 +++++++
 RSDKv2/RetroEngine.cpp  |  4 ++++
 RSDKv2/Ini.cpp          |  2 +-
 RSDKv2/Debug.cpp        |  4 ++++
 RSDKv2/Reader.cpp       |  2 +-
 6 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/RSDKv2/RetroEngine.hpp b/RSDKv2/RetroEngine.hpp
--- a/RSDKv2/RetroEngine.hpp
+++ b/RSDKv2/RetroEngine.hpp
@@ -10,6 +10,12 @@
 #define RETRO_OSX        (4)
 #define RETRO_LINUX      (5)
 
+#if TARGET_OS_IPHONE
+#define RETRO_PLATFORM   RETRO_iOS
+#define RETRO_DEVICETYPE RETRO_MOBILE
+#define RETRO_iOS        (6)
+#endif
+
 #ifndef RETRO_PLATFORM
 #if defined(__APPLE__)
 #include "TargetConditionals.h"
diff --git a/RSDKv2/Userdata.cpp b/RSDKv2/Userdata.cpp
--- a/RSDKv2/Userdata.cpp
+++ b/RSDKv2/Userdata.cpp
@@ -1,5 +1,12 @@
 #include "RetroEngine.hpp"
 
+#if RETRO_PLATFORM == RETRO_iOS
+#import <Foundation/Foundation.h>
+static const char *getDocumentsPath() {
+    static NSString *_docsBuf = nil;
+    if (!_docsBuf) {
+        NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+        _docsBuf = [[paths firstObject] stringByAppendingString:@"/"];
+    }
+    return [_docsBuf UTF8String];
+}
+#endif
+
 void InitUserdata() {
+#if RETRO_PLATFORM == RETRO_iOS
+    StrCopy(gamePath, getDocumentsPath());
+#else
 
diff --git a/RSDKv2/RetroEngine.cpp b/RSDKv2/RetroEngine.cpp
--- a/RSDKv2/RetroEngine.cpp
+++ b/RSDKv2/RetroEngine.cpp
@@ -XX,6 +XX,10 @@
+#elif RETRO_PLATFORM == RETRO_iOS
+    StrCopy(dest, gamePath);
+    StrAdd(dest, Engine.dataFile);
 #elif RETRO_PLATFORM == RETRO_ANDROID
 
diff --git a/RSDKv2/Ini.cpp b/RSDKv2/Ini.cpp
--- a/RSDKv2/Ini.cpp
+++ b/RSDKv2/Ini.cpp
@@ -XX,7 +XX,7 @@
-#elif RETRO_PLATFORM == RETRO_OSX
+#elif RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_iOS
 
diff --git a/RSDKv2/Debug.cpp b/RSDKv2/Debug.cpp
--- a/RSDKv2/Debug.cpp
+++ b/RSDKv2/Debug.cpp
@@ -XX,6 +XX,10 @@
+#elif RETRO_PLATFORM == RETRO_iOS
+    sprintf(pathBuffer, "%s/log.txt", gamePath);
 
diff --git a/RSDKv2/Reader.cpp b/RSDKv2/Reader.cpp
--- a/RSDKv2/Reader.cpp
+++ b/RSDKv2/Reader.cpp
@@ -XX,7 +XX,7 @@
-    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID
+    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_iOS
